builder/swig/go.bzl

```diff
@@ -0,0 +1,201 @@
+    # complain about the go file. The .h is provided in the CcInfo.
+
+    return [
+        DefaultInfo(files = depset([wrap_src])),
+        OutputGroupInfo(
+            go_source = depset([wrap_go_dir]),
+            cxx_source = depset([wrap_src]),
+        ),
+        CcInfo(
+            compilation_context = compilation_context,
+            linking_context = ctx.attr.lib[CcInfo].linking_context,
+        ),
+    ]
+
+
+
+_swig_go_wrapper = rule(
+    implementation = _swig_go_wrapper_impl,
+    attrs = {
+        "lib": attr.label(
+            doc = "The C library for which to generate the wrapper",
+            providers = [CcInfo],
+            mandatory = True,
+        ),
+        "class_name": attr.string(
+            doc = "Optional override for the java class name (default: same as target name)",
+        ),
+        "interface": attr.label(
+            doc = "Optional SWIG interface (.i) file",
+            allow_single_file = True,
+        ),
+        "includes": attr.label_list(
+            doc = "Additional SWIG files required for wrapper generation",
+            allow_files = True,
+        ),
+#        "package": attr.string(
+#            doc = "Java package for which to generate the sources",
+#            mandatory = True,
+#        ),
+        "enable_cxx": attr.bool(
+            doc = "Enable SWIG C++ processing (default: False)",
+            default = False,
+        ),
+        "extra_args": attr.string_list(
+            doc = "Extra arguments to be passed to SWIG",
+        ),
+#        "_jni_header": attr.label(
+#            default = Label("@bazel_tools//tools/jdk:jni_header"),
+#            allow_single_file = True,
+#        ),
+#        "jni_md_header": attr.label(
+#            allow_single_file = True,
+#        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+def swig_go(name, lib, shared_lib_name=None, tags=[], **kwargs):
+    swig_wrapper_name = name + "_swig"
+
+    _swig_go_wrapper(
+        name = swig_wrapper_name,
+        class_name = name,
+        lib = lib,
+        **kwargs,
+    )
+
+    if not shared_lib_name:
+        shared_lib_name = name
+
+#    out_file = ctx.actions.declare_file("foo.go")
+
+#    cmd = """
+#        cat {swig_wrapper_name}/target.go > {out_f}
+#    """.format(out_f = out_file.path)
+
+#    ctx.actions.run_shell(
+#        outputs= [out_file],
+#        command=cmd,
+#        )
+
+    native.filegroup(
+        name = "source_file",
+        srcs = [swig_wrapper_name],
+        output_group = "go_source",
+    )
+
+    # copy the generated go file (assuming it is called typedb_driver.go) into an output file with the same name
+    native.genrule(
+        name="go_wrapper_copier",
+        outs= ["typedb_driver.go"],
+        srcs =[":source_file"],
+        cmd_bash = "cat $(SRCS)/typedb_driver.go > $@",
```

Comment:
Why do you use `cat` to copy files?
If you still use this rule, we also need a version for `windows`.